controller/engine_controller.go

```diff
@@ -1814,6 +1828,58 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 	return nil
 }
 
+// updateReplicaRebuildFailedCondition updates the rebuild failed condition if replica rebuilding failed
+func updateReplicaRebuildFailedCondition(log *logrus.Entry, ds *datastore.DataStore, replicaName, errMsg string) error {
+	replica, err := ds.GetReplica(replicaName)
+	if err != nil {
+		return err
+	}
+
+	replicaRebuildFailedReason, conditionStatus, isFailedByNetwork := getReplicaRebuildFailedReason(errMsg)
+	if isFailedByNetwork {
+		replicaKubernetesNode, err := ds.GetKubernetesNode(replica.Spec.NodeID)
+		if err != nil {
+			return err
+		}
+		for _, con := range replicaKubernetesNode.Status.Conditions {
+			if con.Type != v1.NodeReady {
+				continue
+			}
+			if con.Status != v1.ConditionTrue {
+				replicaRebuildFailedReason = longhorn.ReplicaConditionReasonRebuildFailedNodeDown
+			}
+			break
+		}
+	}
```

Comment:
NIT: You can rename `getReplicaRebuildFailedReason` to `getRebuildFailedReasonFromError`, then have a new function `getReplicaRebuildFailedReason` includes `getRebuildFailedReasonFromError`and this part. You can input the node CR or `ds` for the function

Reply:
Modified. Renamed to `getRebuildFailedReasonFromError` and add a new `getRebuildFailedReason` with getting kubernetes node status